network-libp2p/src/network.rs

```diff
@@ -1717,15 +1635,28 @@ impl Network {
         }
 
         if let Ok(request_id) = output_rx.await {
-            let timeout = Req::TIME_WINDOW.mul_f32(1.5f32);
-            let future = response_rx.map_err(std::io::Error::other);
-            match wasm_timer::TryFutureExt::timeout(future, timeout).await {
+            let result = match timeout(Req::TIME_WINDOW.mul_f32(1.5f32), response_rx).await {
+                Ok(res) => res,
                 Err(_) => {
-                    warn!(%request_id, request_type = std::any::type_name::<Req>(), %peer_id, "Request timed out with no response from libp2p");
-                    Err(RequestError::OutboundRequest(
+                    warn!(
+                        %request_id,
+                        request_type = std::any::type_name::<Req>(),
+                        %peer_id,
+                        "Request timed out with no response from libp2p"
+                    );
+                    return Err(RequestError::OutboundRequest(
                         OutboundRequestError::SenderFutureDropped,
-                    ))
+                    ));
                 }
+            };
+
+            // let timeout = Req::TIME_WINDOW.mul_f32(1.5f32);
+            // let future = response_rx.map_err(std::io::Error::other);
+            //match wasm_timer::TryFutureExt::timeout(future, timeout).await {
+            match result {
+                Err(_) => Err(RequestError::OutboundRequest(
+                    OutboundRequestError::SenderFutureDropped,
+                )),
```

Comment:
Could you add a `warn!` here so the condition could be more easily noticed in logs?